### DISCUSSION TITLE
Speach to text with go SDK

### DISCUSSION THREAD

**Discussion by gesaleh** [deepgram/community#951] [General help] [answered]
I'm trying to make the speach to talk work but i get always issue with response zero words
attached the main_test file and logs i'm gettings
probably i'm missing something or doing it wrong i don't see yet :(
[Archive.zip](https://github.com/user-attachments/files/17275468/Archive.zip)

any help will be appreciated

**Comment by jpvajda** [comment]
@gesaleh  👋  could you provide us with a request ID and the API request including all query params you are try to make?

**Reply by gesaleh** [reply]
any update also i noticed none of my callback functions are triggered

**Comment by deepgram-community** [comment]
any updates @gesaleh?

> luke

**Comment by gesaleh** [comment]
hello please find attached my debugging and parameters they are in the zip file
` "request_id": "76f6730f-5625-43ee-8b40-e4036486fe07"`

```
clientLISTEN.Init(clientLISTEN.InitLib{
		LogLevel: clientLISTEN.LogLevelFull, // LogLevelDefault, LogLevelFull, LogLevelDebug, LogLevelTrace
	})

	// Set the listening options (language, punctuate, etc.)
	// Client options
	cOptions := &interfaces.ClientOptions{
		EnableKeepAlive: true,
	}

	// Transcription options
	tOptions := &interfaces.LiveTranscriptionOptions{
		Model:     "nova-2",
		Language:  "en-US",
		Punctuate: true,
		Encoding:  "linear16",
		//Channels:       1,
		SampleRate:  16000,
		SmartFormat: true,
		//VadEvents:      true,
		//InterimResults: true,
		//UtteranceEndMs: "1000",
	}

	callback := lib.GetCallback()
	//callback := lib.NewSWCallback(conn)
	clientSTT, err = clientLISTEN.NewWSUsingCallback(ctx, os.Getenv("DEEPGRAM_API_KEY"), cOptions, tOptions, callback)
	if err != nil {
		log.Fatalf("Error initializing Deepgram STT client: %v", err)
	}

	// Connect to the Deepgram STT WebSocket
	bConnected := clientSTT.Connect()
	if !bConnected {
		log.Println("Client.Connect failed")
		log.Printf("error starting Deepgram STT WebSocket")
	}
```
i  get always 0 no words 
`2024/10/07 09:49:27 sentence recevied :  &{Channel:{Alternatives:[{Confidence:0 Transcript: Words:[] Languages:[]}]} ChannelIndex:[0 1] Duration:1.024 IsFinal:false FromFinalize:false Metadata:{Extra:map[] ModelInfo:{Arch:nova-2 Name:2-general-nova Version:2024-01-18.26916} ModelUUID:c0d1a568-ce81-4fea-97e7-bd45cb1fdf3c RequestID:76f6730f-5625-43ee-8b40-e4036486fe07} SpeechFinal:false Start:0 Type:Results}`

i shared the callback module and main.go also in the archive.zip file

**Comment by deepgram-community** [comment, marked as answer]
According to our systems, in the 14 minutes you were connected, we received 1.308 seconds of audio data.

> luke
